structuretoolkit/analyse/snap.py

```diff
@@ -0,0 +1,601 @@
+        cutoff (float): cutoff radius for the construction of the neighbor list
+
+    Returns:
+        np.ndarray: Numpy array with the calculated descriptor derivatives
+    """
+    lmp, bispec_options, cutoff = _get_default_parameters(
+        atom_types=atom_types,
+        twojmax=twojmax,
+        element_radius=element_radius,
+        rcutfac=rcutfac,
+        rfac0=rfac0,
+        rmin0=rmin0,
+        bzeroflag=bzeroflag,
+        quadraticflag=quadraticflag,
+        weights=weights,
+        cutoff=cutoff,
+    )
+    return _calc_snap_per_atom(
+        lmp=lmp, structure=structure, bispec_options=bispec_options, cutoff=cutoff
+    )
+
+
+def calc_snap_descriptor_derivatives(
+    structure,
+    atom_types,
+    twojmax=6,
+    element_radius=4.0,
+    rcutfac=1.0,
+    rfac0=0.99363,
+    rmin0=0.0,
+    bzeroflag=False,
+    quadraticflag=False,
+    weights=None,
+    cutoff=10.0,
+):
+    """
+    Calculate per atom derivatives of the SNAP descriptors using LAMMPS https://docs.lammps.org/compute_sna_atom.html
+
+    Args:
+        structure (ase.atoms.Atoms): atomistic structure as ASE atoms object
+        atom_types (list): list of element types
+        twojmax (int): band limit for bispectrum components (non-negative integer)
+        element_radius (list): list of radii for the individual elements
+        rcutfac (float): scale factor applied to all cutoff radii (positive real)
+        rfac0 (float): parameter in distance to angle conversion (0 < rcutfac < 1)
+        rmin0 (float): parameter in distance to angle conversion (distance units)
+        bzeroflag (bool): subtract B0
+        quadraticflag (bool): generate quadratic terms
+        weights (list/np.ndarry/None): list of neighbor weights, one for each type
+        cutoff (float): cutoff radius for the construction of the neighbor list
+
+    Returns:
+        np.ndarray: Numpy array with the calculated descriptor derivatives
+    """
+    lmp, bispec_options, cutoff = _get_default_parameters(
+        atom_types=atom_types,
+        twojmax=twojmax,
+        element_radius=element_radius,
+        rcutfac=rcutfac,
+        rfac0=rfac0,
+        rmin0=rmin0,
+        bzeroflag=bzeroflag,
+        quadraticflag=quadraticflag,
+        weights=weights,
+        cutoff=cutoff,
+    )
+    return _calc_snap_derivatives(
+        lmp=lmp, structure=structure, bispec_options=bispec_options, cutoff=cutoff
+    )
+
+
+def get_snap_descriptor_names(twojmax):
+    """
+    Get names of the SNAP descriptors
+
+    Args:
+        twojmax (int): band limit for bispectrum components (non-negative integer)
+
+    Returns:
+        list: List of SNAP descriptor names
+    """
+    lst = []
+    for j1 in range(0, twojmax + 1):
+        for j2 in range(0, j1 + 1):
+            for j in range(j1 - j2, min(twojmax, j1 + j2) + 1, 2):
+                if j >= j1:
+                    lst.append([j1 / 2.0, j2 / 2.0, j / 2.0])
```

Comment:
I already commented on this in my last review.

```python
for j in range(j1 - j2, foo, bar):
    if j >= j1:
        do_something()
    else:
        do_nothing()
```

Is functionally equivalent to 

```
for j in range(j1, foo, bar):
        do_something()
```

Except that the former is both more code _and_ less efficient. Please fix this.

Reply:
The increment is `2` rather than `1`, so just starting at `j1` is not sufficient.